### PR TITLE
[#7707] Clarify groupadmin behavior with test (main)

### PR DESCRIPTION
**Pull request by MartinFlores751**
Adds a test to clarify expected groupadmin behavior when misusing igroupadmin.

Based on the comments linked in the issue, it seems that one test may be enough, though if there are some other cases, suggestions are encouraged.

### CODE REVIEW

**Review by alanking — APPROVED** [review]
Looks great! Let's get one more approval, and I say this is good to go.

I can't think of any additional tests which would better fulfill the requirement set out by the issue.

File: scripts/irods/test/test_igroupadmin.py [hunk 1]
```diff
@@ -142,6 +142,26 @@ def test_group_administrator_scenarios__issue_6888(self):
...
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkgroup', self.group])
+            self.groupadmin.assert_icommand(['igroupadmin', 'mkuser', test_username])
+
+            # Populate empty group. groupadmin should no longer have any control of this group now
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, test_username])
+
+            # Assert groupadmin cannot control this group now
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'atg', self.group, self.groupadmin.username])
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'rfg', self.group, test_username])
+            self.groupadmin.assert_icommand_fail(['igroupadmin', 'lg', self.group])
```
Comment (korydraughn) [inline scripts/irods/test/test_igroupadmin.py:158]: What iRODS error codes do these commands print out?
I'm thinking we should assert the iRODS error codes too. Thoughts?

**Review by korydraughn — COMMENTED** [review]
Clean and straight to the point.

File: scripts/irods/test/test_igroupadmin.py (hunk 1 above)
Comment (alanking) [reply scripts/irods/test/test_igroupadmin.py:158]: That's fair. Hopefully something along the lines of `CAT_NO_ACCESS_PERMISSION`...

Comment (MartinFlores751) [reply scripts/irods/test/test_igroupadmin.py]: Done.

File: scripts/irods/test/test_igroupadmin.py [hunk 2]
```diff
@@ -149,13 +149,15 @@ def test_groupadmin_can_lock_self_out_of_group__issue_7707(self):
...
+            # Populate empty group. groupadmin should no longer have any control of the group now
             self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, test_username])
 
-            # Assert groupadmin cannot control this group now
-            self.groupadmin.assert_icommand_fail(['igroupadmin', 'atg', self.group, self.groupadmin.username])
-            self.groupadmin.assert_icommand_fail(['igroupadmin', 'rfg', self.group, test_username])
-            self.groupadmin.assert_icommand_fail(['igroupadmin', 'lg', self.group])
+            # Assert groupadmin cannot control the group
+            self.groupadmin.assert_icommand(['igroupadmin', 'atg', self.group, self.groupadmin.username], 'STDERR', 'CAT_INSUFFICIENT_PRIVILEGE_LEVEL', desired_rc=0)
+            self.groupadmin.assert_icommand(['igroupadmin', 'rfg', self.group, test_username], 'STDERR', 'CAT_INSUFFICIENT_PRIVILEGE_LEVEL', desired_rc=0)
```
Comment (korydraughn) [inline scripts/irods/test/test_igroupadmin.py:157]: To future proof this even more, include the iRODS error code with the symbolic name. That way, if `<prefix>CAT_INSUFFICIENT_PRIVILEGE_LEVEL<suffix>` is returned, the test won't be tripped up by it.

**Review by korydraughn — COMMENTED** [review]
Once the integer error codes are included, I think we'll be ready for squashing.

File: scripts/irods/test/test_igroupadmin.py (hunk 2 above)
Comment (MartinFlores751) [reply scripts/irods/test/test_igroupadmin.py]: Done.

**Comment by korydraughn** [comment]
Squash it.

**Comment by MartinFlores751** [comment]
Squashed.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by MartinFlores751** [comment]
Pounded.

**Comment by korydraughn** [comment]
Please cherry-pick to 4-3-stable.

**Pull request merged**